src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -555,17 +596,55 @@ private void transferAlignmentInfoToFragment(final SAMRecord unaligned, final SA
             log.warn("Record mapped off end of reference; making unmapped: " + aligned);
             SAMUtils.makeReadUnmapped(unaligned);
         } else if (isContaminant) {
-            log.warn("Record looks like foreign contamination; making unmapped: " + aligned);
-            // NB: for reads that look like contamination, just set unmapped flag and zero MQ but keep other flags as-is.
-            // this maintains the sort order so that downstream analyses can use them for calculating evidence
-            // of contamination vs other causes (e.g. structural variants)
+            if (numCrossSpeciesContaminantWarnings < MAX_CROSS_SPECIES_CONTAM_WARN) {
+                log.warn("Record looks like cross-species contamination ; unmapping read: " + aligned);
+                numCrossSpeciesContaminantWarnings++;
+            } else if (numCrossSpeciesContaminantWarnings == MAX_CROSS_SPECIES_CONTAM_WARN) {
+                log.warn("No more warnings regarding cross-species contamination; limit reached.");
+            }
+
+            if (unmappingReadsStrategy.isPopulatePATag()) {
+                unaligned.setAttribute("PA", encodeMappingInformation(aligned));
+            }
+
+
+            if (unmappingReadsStrategy.isResetMappingInformation()) {
+                unaligned.setReferenceIndex(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX);
+                unaligned.setAlignmentStart(0);
+                unaligned.setCigar(null);
+                unaligned.setCigarString(SAMRecord.NO_ALIGNMENT_CIGAR);
```

Comment:
I was thinking that it would just null out those attributes.  But I see your point that other values are placed into your `PA` tag.